docs: CLI documentation standardization (#16563)
eslint/eslint: docs/src/user-guide/command-line-interface.md

**Note:** You can also use alternative package managers such as [Yarn](https://yarnpkg.com/) or [pnpm](https://pnpm.io/) to run ESLint. Please refer to your package manager's documentation for the correct syntax.

> ## Pass multiple values to an option

Reviewer: ```suggestion
## Pass Multiple Values to an Option
```

**Note:** You can also use alternative package managers such as [Yarn](https://yarnpkg.com/) or [pnpm](https://pnpm.io/) to run ESLint. Please refer to your package manager's documentation for the correct syntax.

## Pass multiple values to an option

Options that accept multiple values can be specified by repeating the option or with a comma-delimited list (other than [`--ignore-pattern`](#--ignore-pattern), which does not allow the second style).

Examples of options that accept multiple values:

```shell
npx eslint --ext .jsx --ext .js lib/
# OR
npx eslint --ext .jsx,.js lib/
```

## Options

You can view all the CLI options by running `npx eslint -h`.

```txt
eslint [options] file.js [file.js] [dir]

Basic configuration:

>   --no-config-lookup             Disable look up for eslint.config.js

Reviewer 2: This is the output of `-h `in flat config mode, but this document should still be about the eslintrc config mode. To switch back to the eslintrc mode, you can set `ESLINT_USE_FLAT_CONFIG=false` environment variable, and then run `npx eslint -h` again to get the help output in eslintrc config mode.

**Note:** You can also use alternative package managers such as [Yarn](https://yarnpkg.com/) or [pnpm](https://pnpm.io/) to run ESLint. Please refer to your package manager's documentation for the correct syntax.

## Options

The command line utility has several options. You can view the options by running `npx eslint -h`.

```text
eslint [options] file.js [file.js] [dir]

Basic configuration:
  --no-eslintrc                   Disable use of configuration from .eslintrc.*
  -c, --config path::String       Use this configuration, overriding .eslintrc.* config options if present
  --env [String]                  Specify environments
  --ext [String]                  Specify JavaScript file extensions
  --global [String]               Define global variables
  --parser String                 Specify the parser to be used
  --parser-options Object         Specify parser options

>   --resolve-plugins-relative-to path::String  A folder where plugins should be resolved from, CWD by default

Reviewer 2: I'm not sure if we want to remove this overview of all options.

(this is the output from `eslint -h`)

Author: i removed b/c user can just run the command to get this. i don't think there's value in documenting the output. also if the command this would also have to changed or become out of date. 

i think having it provides a non-trivial maintenance burden for little value.

Reviewer 3: I also feel like we can have this

Author: added back w refactored intro text in [78affe7](https://github.com/eslint/eslint/pull/16563/commits/78affe73a22805015a9af1a16b1ab6e755a8f616)

#### `--no-eslintrc`

Disables use of configuration from `.eslintrc.*` and `package.json` files.

* **Argument Type**: No argument.

##### `--no-eslintrc` example

```shell
npx eslint --no-eslintrc file.js
```

> #### `--config`

Author: resolved in [288918e](https://github.com/eslint/eslint/pull/16563/commits/288918e12e2109f86ea34f2ec72e523d2a15be54)

This option allows you to specify an additional configuration file for ESLint (see [Configuring ESLint](configuring/) for more).

* **Alias**: `-c`

> * **Argument Type**: String. File name.

Reviewer 2: ```suggestion
* **Argument Type**: String. Path to file.
```

#### `--ext`

This option allows you to specify which file extensions ESLint uses when searching for target files in the directories you specify.

* **Argument Type**: String. File extension.
* **Multiple Arguments**: Yes
* **Default Value**: `.js` and the files that match the `overrides` entries of your configuration.

> `--ext` is only used when the arguments are directories. If you use glob patterns or file names, then `--ext` is ignored. For example, `npx eslint lib/* --ext .js` matches all files within the `lib/` directory, regardless of extension.

Author: Resolved in [8087577](https://github.com/eslint/eslint/pull/16563/commits/8087577692d71323e90ced8c8e3d81ec61d8e93b)

#### `--ext`

This option allows you to specify which file extensions ESLint uses when searching for target files in the directories you specify.

> * **Argument Type**: String. File glob pattern.

Reviewer 2: Expected format is `.<extension>` (for example, `.ts`), I think this shouldn't be described as "File glob pattern".

Author: fixed in 89113ac71e2f71d509c499b3d5883e643228e149

* **Multiple Arguments**: Yes

> * **Default Value**: `*.js` and the files that match the `overrides` entries of your configuration.

Reviewer 2: ```suggestion
* **Default Value**: `.js` and the files that match the `overrides` entries of your configuration.
```

npx eslint . --ext .js,.ts
```

#### `--global`

This option defines global variables so that they are not  flagged as undefined by the [`no-undef`](../rules/no-undef) rule.

> * **Argument Type**: String. Name of global variable. Any specified global variables are assumed to be read-only by default, but appending `:true` to a variable's name ensures that `no-undef` also allows writes.

Reviewer 4: ```suggestion
* **Argument Type**: String. Name of the global variable. Any specified global variables are assumed to be read-only by default, but appending `:true` to a variable's name ensures that `no-undef` also allows writes.
```

Disables use of configuration from `.eslintrc.*` and `package.json` files.

* **Argument Type**: No argument.

##### `--no-eslintrc` example

```shell
npx eslint --no-eslintrc file.js
```

> #### `--config`, `-c`

Reviewer 2: Perhaps we should keep `` `-c`, `--config` `` (and others) in that order to match the output of `eslint -h` where short flags come first?

Reviewer: Agree

Author: ok, switching order

#### `--parser`

This option allows you to specify a parser to be used by ESLint.

* **Argument Type**: String. Parser to be used by ESLint.
* **Multiple Arguments**: No
* **Default Value**: `espree`

##### `--parser` example

```shell
# Use TypeScript ESLint parser
npx eslint --parser @typescript-eslint/parser files.ts
```

#### `--parser-options`

This option allows you to specify parser options to be used by ESLint. The available parser options are determined by the parser being used.

* **Argument Type**: Key/Value pair. Separated by colon (`:`).
* **Multiple Arguments**: Yes

##### `--parser-options` example

```shell
echo '3 ** 4' | npx eslint --stdin --parser-options ecmaVersion:6 # fails with a parsing error
echo '3 ** 4' | npx eslint --stdin --parser-options ecmaVersion:7 # succeeds, yay!
```

#### `--resolve-plugins-relative-to`

Changes the folder where plugins are resolved from.

> * **Argument Type**: File path.

Reviewer 4: ```suggestion
* **Argument Type**: String. Path to file.
```

* **Multiple Arguments**: No
* **Default Value**: By default, plugins are resolved from the current working directory.

This option should be used when plugins were installed by someone other than the end user. It should be set to the project directory of the project that has a dependency on the necessary plugins. If you use npm to install plugins, the path ends with `/node_modules/`.

For example:

* When using a config file that is located outside of the current project (with the `--config` flag), if the config uses plugins which are installed locally to itself, `--resolve-plugins-relative-to` should be set to the directory containing the config file.
* If an integration has dependencies on ESLint and a set of plugins, and the tool invokes ESLint on behalf of the user with a preset configuration, the tool should set `--resolve-plugins-relative-to` to the top-level directory of the tool.

##### `--resolve-plugins-relative-to` example

> ```sh

Reviewer 2: ```suggestion
```shell
```

This should fix the site build error.

```
9:34:06 PM: [11ty] 1. Having trouble rendering njk template ./src/user-guide/command-line-interface.md (via TemplateContentRenderError)
9:34:06 PM: [11ty] 2. "sh" is not a valid Prism.js language for eleventy-plugin-syntaxhighlight (via Error)
```

npx eslint --config ~/personal-eslintrc.js \
--resolve-plugins-relative-to /usr/local/lib/node_modules/
```

> ### Specify rules and plugins

Reviewer 2: If we want to rename group headings, I think we should do the same in [`lib/options.js`](https://github.com/eslint/eslint/blob/f5808cb51529174a67b4938223f06435ad6d5118/lib/options.js#L180) so that the output of `eslint -h` is consistent with the CLI reference documentation.

Author: didn't realize there was congruency btwn this page and the headers there. i'll update those strings as part of this PR.

Author: i tried to add the updates to the strings in `lib/options.js`, but the pre-commit hook is failing with the following error. any ideas on how to fix?

```
 > running pre-commit hook: lint-staged
✔ Preparing...
⚠ Running tasks...
  ❯ Running tasks for *.js
    ✖ eslint --fix [FAILED]
  ↓ No staged files match *.md [SKIPPED]
  ↓ No staged files match docs/src/rules/*.md [SKIPPED]
  ↓ No staged files match docs/**/*.svg [SKIPPED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up...

✖ eslint --fix:

Oops! Something went wrong! :(

ESLint: 8.28.0

Error: Cannot find module 'is-path-inside'
Require stack:
- /Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/lib/eslint/eslint-helpers.js
- /Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/lib/eslint/flat-eslint.js
- /Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/lib/eslint/index.js
- /Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/lib/cli.js
- /Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/bin/eslint.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:999:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/benperlmutter1/Library/Mobile Documents/com~apple~CloudDocs/eslint/lib/eslint/eslint-helpers.js:21:22)
    at Module._compile (node:internal/modules/cjs/loader:1097:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1149:10)
    at Module.load (node:internal/modules/cjs/loader:975:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:999:19)

pre-commit hook failed (add --no-verify to bypass)
```

Reviewer 2: You probably have outdated `node_modules`, so they don't contain all the currently needed dependencies. Try removing the `node_modules` directory and then run `npm install` to generate it again.

Author: yep ,that did it. updated the help output text in [19117e9](https://github.com/eslint/eslint/pull/16563/commits/19117e964a93068d0c86a85197b81a53988b6fc6)

#### `--env`

This option enables specific environments.

* **Argument Type**: String. One of the available environments.
* **Multiple Arguments**: Yes

> Details about the global variables defined by each environment are available in the [Specifying Environments](configuring/language-options#specifying-environments) documentation. This option only enables environments. It does not disable environments set in other configuration files. To specify multiple environments, separate them using commas, or use the option multiple times.

Reviewer: Just double checking -- specifying vs. specify in the link?

Author: yep, this is currently the name of the section

#### `--parser`

This option allows you to specify a parser to be used by ESLint.

* **Argument Type**: String. Parser to be used by ESLint.
* **Multiple Arguments**: No
* **Default Value**: `espree`

##### `--parser` example

```shell
# Use TypeScript ESLint parser
npx eslint --parser @typescript-eslint/parser files.ts
```

#### `--parser-options`

This option allows you to specify parser options to be used by ESLint. The available parser options are determined by the parser being used.

* **Argument Type**: Key/value pair separated by colon (`:`).
* **Multiple Arguments**: Yes

##### `--parser-options` example

```shell
echo '3 ** 4' | npx eslint --stdin --parser-options ecmaVersion:6 # fails with a parsing error
echo '3 ** 4' | npx eslint --stdin --parser-options ecmaVersion:7 # succeeds, yay!
```

#### `--resolve-plugins-relative-to`

Changes the folder where plugins are resolved from.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

> * **Default Value**: By default, plugins are resolved from the current working directory.

Reviewer: ```suggestion
* **Default Value**: By default, plugins are resolved from the directory in which your configuration file is found.
```

This option should be used when plugins were installed by someone other than the end user. It should be set to the project directory of the project that has a dependency on the necessary plugins. If you use npm to install plugins, the path ends with `/node_modules/`.

For example:

* When using a config file that is located outside of the current project (with the `--config` flag), if the config uses plugins which are installed locally to itself, `--resolve-plugins-relative-to` should be set to the directory containing the config file.
* If an integration has dependencies on ESLint and a set of plugins, and the tool invokes ESLint on behalf of the user with a preset configuration, the tool should set `--resolve-plugins-relative-to` to the top-level directory of the tool.

##### `--resolve-plugins-relative-to` example

```shell
npx eslint --config ~/personal-eslintrc.js \
--resolve-plugins-relative-to /usr/local/lib/node_modules/
```

> ### Specify rules and plugins

Reviewer: ```suggestion
### Specify Rules and Plugins
```

#### `--rule`

> This option specifies rules to be used.

Reviewer 4: ```suggestion
This option specifies the rules to be used.
```

* **Argument Type**: Rules and their configuration specified with [levn](https://github.com/gkz/levn#levn--) format.
* **Multiple Arguments**: Yes

> These rules are merged with any rules specified with configuration files. If the rule is defined in a plugin, you have to prefix the rule ID with the plugin name and a `/`.

Reviewer 2: Perhaps we should keep the note about `--no-eslintrc` here?

Author: i think i removed the line b/c i didn't understand what it was saying, and figured it wasn't too important b/c it's in parenthesis. 

could you clarify what is meant by "You can use `--no-eslintrc` to change that behavior."? then i can add back in refactored, clearer form.

Reviewer: Agree with @bpmutter, I think it’s safe to omit this.

Reviewer 2: This was added in https://github.com/eslint/eslint/pull/2776 per https://github.com/eslint/eslint/issues/2773.

It explains how to run only the rules specified on the command line. Without `--no-eslintrc`, rules specified with config files will be run too. With `--no-eslintrc`, rules specified with `--rule` are not "merged with any rules specified with configuration files", meaning that the final list of rules to run is the list of rules specified with `--rule`.

I'm fine with removing the note, but it also makes sense to keep it here as it addresses one of the common use cases for `--rule`.

Reviewer: I think @bpmutter's point, and I agree, is that the current text isn't clear as to what `--no-eslintrc` does in this context. It's way to vague to be able to understand your point. We can either remove the parens and leave it, or we can update the text to more explicitly describe the situation you mention, but I don't think leaving as-is makes sense.

Author: ok, thank you for the further clarification! 

i added a sentence to explain and example in 0365e98704475189f617e69c6bd000ae0b94e625

This option allows you to specify another directory from which to load rules files. This allows you to dynamically load new rules at run time. This is useful when you have custom rules that aren't suitable for being bundled with ESLint.

> The rules in your custom rules directory must follow the same format as bundled rules to work properly. You can also specify multiple locations for custom rules by including multiple `--rulesdir` options.

Reviewer 2: `--rulesdir` is missing "Argument Type" and "Multiple Arguments: No"

Author: sure, added in 4e49c050d

##### `--fix` example

```shell
npx eslint --fix file.js
```

#### `--fix-dry-run`

This option has the same effect as `--fix` with the difference that the fixes are not saved to the file system. Because the default formatter does not output the fixed code, you'll have to use another formatter (e.g. `--format json`) to get the fixes.

* **Argument Type**: No argument.

This makes it possible to fix code from `stdin` when used with the `--stdin` flag.

This flag can be useful for integrations (e.g. editor plugins) which need to autofix text from the command line without saving it to the filesystem.

##### `--fix-dry-run` example

```shell
getSomeText | npx eslint --stdin --fix-dry-run --format json
```

#### `--fix-type`

This option allows you to specify the type of fixes to apply when using either `--fix` or `--fix-dry-run`.

* **Argument Type**: String. One of the following fix types:
  1. `problem` - fix potential errors in the code
  1. `suggestion` - apply fixes to the code that improve it
  1. `layout` - apply fixes that do not change the program structure (AST)
  1. `directive` - apply fixes to inline directives such as `// eslint-disable`
* **Multiple Arguments**: Yes

> This option is helpful if you are using another program to format your code but you would still like ESLint to apply other types of fixes.

Reviewer 4: ```suggestion
This option is helpful if you are using another program to format your code but would still like ESLint to apply other types of fixes.
```

Author: ```suggestion
This option is helpful if you are using another program to format your code, but you would still like ESLint to apply other types of fixes.
```

i think adding comma is better fix here. clears up parts of the sentence

> Changes the folder where plugins are resolved from.

Reviewer: For consistency, we should probably just stick with "directory" instead of bouncing back and forth with "folder"

```suggestion
Changes the directory where plugins are resolved from.
```

This option allows you to specify another directory from which to load rules files. This allows you to dynamically load new rules at run time. This is useful when you have custom rules that aren't suitable for being bundled with ESLint.

* **Argument Type**: String. Path to directory. The rules in your custom rules directory must follow the same format as bundled rules to work properly.
* **Multiple Arguments**: Yes.

Note that, as with core rules and plugin rules, you still need to enable the rules in configuration or via the `--rule` CLI option in order to actually run those rules during linting. Specifying a rules directory with `--rulesdir` does not automatically enable the rules within that directory.

##### `--rulesdir` example

```shell
npx eslint --rulesdir my-rules/ file.js
npx eslint --rulesdir my-rules/ --rulesdir my-other-rules/ file.js
```

> ### Fix problems

Reviewer: ```suggestion
### Fix Problems
```

* **Argument Type**: No argument.

This makes it possible to fix code from `stdin` when used with the `--stdin` flag.

This flag can be useful for integrations (e.g. editor plugins) which need to autofix text from the command line without saving it to the filesystem.

##### `--fix-dry-run` example

```shell
getSomeText | npx eslint --stdin --fix-dry-run --format json
```

#### `--fix-type`

This option allows you to specify the type of fixes to apply when using either `--fix` or `--fix-dry-run`.

* **Argument Type**: String. One of the following fix types:
  1. `problem` - fix potential errors in the code
  1. `suggestion` - apply fixes to the code that improve it
  1. `layout` - apply fixes that do not change the program structure (AST)
  1. `directive` - apply fixes to inline directives such as `// eslint-disable`
* **Multiple Arguments**: Yes

This option is helpful if you are using another program to format your code, but you would still like ESLint to apply other types of fixes.

##### `--fix-type` example

```shell
npx eslint --fix --fix-type suggestion .
npx eslint --fix --fix-type suggestion --fix-type problem .
npx eslint --fix --fix-type suggestion,layout .
```

> ### Ignore files

Reviewer: ```suggestion
### Ignore Files
```

#### `--no-ignore`

> Disables excluding of files from `.eslintignore` files, `--ignore-path` flags, `--ignore-pattern` flags, and  the `ignorePatterns` property in config files.

Reviewer 4: ```suggestion
Disables excluding of files from `.eslintignore` files, `--ignore-path` flags, `--ignore-pattern` flags, and the `ignorePatterns` property in config files.
```

* **Argument Type**: No argument.

##### `--no-ignore` example

```shell
npx eslint --no-ignore file.js
```

#### `--ignore-pattern`

This option allows you to specify patterns of files to ignore (in addition to those in `.eslintignore`).

> * **Argument Type**: String. The supported syntax is the same as for `.eslintignore` [files](configuring/ignoring-code#the-eslintignore-file), which use the same patterns as the [`.gitignore` specification](https://git-scm.com/docs/gitignore). You should quote your patterns in order to avoid shell interpretation of glob patterns.

Reviewer 4: ```suggestion
* **Argument Type**: String. The supported syntax is the same as for `.eslintignore` [files](configuring/ignoring-code#the-eslintignore-file), which uses the same patterns as the [`.gitignore` specification](https://git-scm.com/docs/gitignore). You should quote your patterns in order to avoid shell interpretation of glob patterns.
```

Author: ```suggestion
* **Argument Type**: String. The supported syntax is the same as for [`.eslintignore` files](configuring/ignoring-code#the-eslintignore-file), which use the same patterns as the [`.gitignore` specification](https://git-scm.com/docs/gitignore). You should quote your patterns in order to avoid shell interpretation of glob patterns.
```

correct as is b/c referring to .eslintignore files (plural). updating link here tho.

* **Multiple Arguments**: Yes
* **Default Value**: whatever the default value is OR N/A

##### `--ignore-pattern` example

```shell
npx eslint --ignore-pattern "/lib/" --ignore-pattern "/src/vendor/*" .
```

### Use stdin

#### `--stdin`

This option tells ESLint to read and lint source code from STDIN instead of from files. You can use this to pipe code to ESLint.

* **Argument Type**: No argument.

##### `--stdin` example

```shell
cat myfile.js | npx eslint --stdin
```

#### `--stdin-filename`

This option allows you to specify a filename to process STDIN as.

> * **Argument Type**: String. File name.

Reviewer 2: ```suggestion
* **Argument Type**: String. Path to file.
```

#### `--no-ignore`

Disables excluding of files from `.eslintignore` files, `--ignore-path` flags, `--ignore-pattern` flags, and the `ignorePatterns` property in config files.

* **Argument Type**: No argument.

##### `--no-ignore` example

```shell
npx eslint --no-ignore file.js
```

#### `--ignore-pattern`

This option allows you to specify patterns of files to ignore (in addition to those in `.eslintignore`).

* **Argument Type**: String. The supported syntax is the same as for [`.eslintignore` files](configuring/ignoring-code#the-eslintignore-file), which use the same patterns as the [`.gitignore` specification](https://git-scm.com/docs/gitignore). You should quote your patterns in order to avoid shell interpretation of glob patterns.
* **Multiple Arguments**: Yes

> * **Default Value**: whatever the default value is OR N/A

Reviewer 2: This is TODO?

Reviewer: ```suggestion
* **Default Value**: None.
```

Author: this line shouldn't have been included. sorry! removed

> * **Default Value**: whatever the default value is OR N/A

Author: ```suggestion
```

##### `--ignore-pattern` example

```shell
npx eslint --ignore-pattern "/lib/" --ignore-pattern "/src/vendor/*" .
```

### Use stdin

#### `--stdin`

This option tells ESLint to read and lint source code from STDIN instead of from files. You can use this to pipe code to ESLint.

* **Argument Type**: No argument.

##### `--stdin` example

```shell
cat myfile.js | npx eslint --stdin
```

#### `--stdin-filename`

This option allows you to specify a filename to process STDIN as.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

This is useful when processing files from STDIN and you have rules which depend on the filename.

##### `--stdin-filename` example

```shell
cat myfile.js | npx eslint --stdin --stdin-filename myfile.js
```

> ### Handle warnings

Reviewer: ```suggestion
### Handle Warnings
```

This option allows you to specify a warning threshold, which can be used to force ESLint to exit with an error status if there are too many warning-level rule violations in your project.

* **Argument Type**: Integer. The maximum number of warnings to allow. To prevent this behavior, do not use this option or specify `-1` as the argument.
* **Multiple Arguments**: No

Normally, if ESLint runs and finds no errors (only warnings), it exits with a success exit status. However, if `--max-warnings` is specified and the total warning count is greater than the specified threshold, ESLint exits with an error status.

##### `--max-warnings` example

```shell
npx eslint --max-warnings 10 file.js
```

### Output

#### `--output-file`, `-o`

Write the output of linting results to a specified file.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

> ##### `--output-file` example

Reviewer: ```suggestion
##### `--output-file`, `-o` example
```

This option allows you to specify a warning threshold, which can be used to force ESLint to exit with an error status if there are too many warning-level rule violations in your project.

> * **Argument Type**: Integer. Maximum number of warnings to allow. To prevent this behavior, do not use this option or specify `-1` as the argument.

Reviewer 4: ```suggestion
* **Argument Type**: Integer. The maximum number of warnings to allow. To prevent this behavior, do not use this option or specify `-1` as the argument.
```

* **Multiple Arguments**: No

Normally, if ESLint runs and finds no errors (only warnings), it exits with a success exit status. However, if `--max-warnings` is specified and the total warning count is greater than the specified threshold, ESLint exits with an error status.

##### `--max-warnings` example

```shell
npx eslint --max-warnings 10 file.js
```

### Output

#### `--output-file`

> Write output of linting results to a specified file.

Reviewer 4: ```suggestion
Write the output of linting results to a specified file.
```

* **Alias**: `-o`
* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

##### `--output-file` example

```shell
npx eslint -o ./test/test.html
```

#### `--format`

This option specifies the output format for the console.

* **Alias**: `-f`

> * **Argument Type**: String. One of the [built-in formatters](formatters) or a custom formatter.

Reviewer 2: ```suggestion
* **Argument Type**: String. One of the [built-in formatters](formatters/) or a custom formatter.
```

Author: the link as it is seems to work fine. any reason to update this?

Reviewer 2: So that internal links always point to canonical URLs. For the formatters page, it's: https://eslint.org/docs/latest/user-guide/formatters/.

https://eslint.org/docs/latest/user-guide/formatters should actually redirect to https://eslint.org/docs/latest/user-guide/formatters/, but that seems impossible to configure on Netlify when proxying the content. We've managed to make the pages work mostly well when opened through non-canonical URLs, but it's still preferable to direct users to canonical ones.

> **Multiple Arguments**: No

Reviewer 2: ```suggestion
* **Multiple Arguments**: No
```

This option allows you to specify a warning threshold, which can be used to force ESLint to exit with an error status if there are too many warning-level rule violations in your project.

* **Argument Type**: Integer. The maximum number of warnings to allow. To prevent this behavior, do not use this option or specify `-1` as the argument.
* **Multiple Arguments**: No

Normally, if ESLint runs and finds no errors (only warnings), it exits with a success exit status. However, if `--max-warnings` is specified and the total warning count is greater than the specified threshold, ESLint exits with an error status.

##### `--max-warnings` example

```shell
npx eslint --max-warnings 10 file.js
```

### Output

#### `--output-file`, `-o`

Write the output of linting results to a specified file.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

##### `--output-file`, `-o` example

```shell
npx eslint -o ./test/test.html
```

#### `--format`, `-f`

This option specifies the output format for the console.

* **Argument Type**: String. One of the [built-in formatters](formatters/) or a custom formatter.

> **Multiple Arguments**: No

Reviewer 2: ```suggestion
* **Multiple Arguments**: No
```

Missing `*` for list item.

![image](https://user-images.githubusercontent.com/44349756/204539938-99faa9a4-b10b-4347-afda-861d1c0bc6bd.png)

#### `-o`, `--output-file`

Write the output of linting results to a specified file.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

##### `-o`, `--output-file` example

```shell
npx eslint -o ./test/test.html
```

#### `-f`, `--format`

This option specifies the output format for the console.

* **Argument Type**: String. One of the [built-in formatters](formatters/) or a custom formatter.
* **Multiple Arguments**: No
* **Default Value**: [`stylish`](formatters/#stylish)

If you are using a custom formatter defined in a local file, you can specify the path to the custom formatter file.

An npm-installed formatter is resolved with or without `eslint-formatter-` prefix.

When specified, the given format is output to the console. If you'd like to save that output into a file, you can do so on the command line like so:

```shell
# Saves the output into the `results.txt` file.
npx eslint -f compact file.js > results.txt
```

##### `-f`, `--format` example

Use the built-in `compact` formatter:

```shell
npx eslint --format compact file.js
```

Use a local custom formatter:

```shell
npx eslint -f ./customformat.js file.js
```

> Use npm-installed formatter:

Reviewer: ```suggestion
Use an npm-installed formatter:
```

```shell
npm install eslint-formatter-pretty

> # THEN

Reviewer: I think this makes it read like you should do the first two lines OR the third, which is incorrect.

Author: i'll add some clarification comments to the code example

Author: added in 801544d8f8dfacfe3a27196d6eea56636917fa39

npx eslint -f pretty file.js
# OR
npx eslint -f eslint-formatter-pretty file.js
```

#### `--color` and `--no-color`

> These options forces the enabling/disabling of colorized output.

Reviewer 2: ```suggestion
These options force the enabling/disabling of colorized output.
```

Autofixed files are not placed in the cache. Subsequent linting that does not trigger an autofix will place it in the cache.

##### `--cache` example

```shell
npx eslint --cache file.js
```

#### `--cache-file`

**Deprecated**: Use `--cache-location` instead.

Path to the cache file. If none specified `.eslintcache` is used. The file is created in the directory where the `eslint` command is executed.

#### `--cache-location`

Specify the path to the cache location. Can be a file or a directory.

* **Argument Type**: String. Path to file or directory. If a directory is specified, a cache file is created inside the specified folder. The name of the file is based on the hash of the current working directory, e.g.: `.cache_hashOfCWD`.
* **Multiple Arguments**: No
* **Default Value**: If no location is specified, `.eslintcache` is used. The file is created in the directory where the `eslint` command is executed.

> If the directory for the cache does not exist make sure you add a trailing `/` on \*nix systems or `\` on Windows. Otherwise the path is assumed to be a file.

Reviewer 4: ```suggestion
If the directory for the cache does not exist make sure you add a trailing `/` on \*nix systems or `\` on Windows. Otherwise, the path is assumed to be a file.
```

##### `--cache-location` example

```shell
npx eslint "src/**/*.js" --cache --cache-location "/Users/user/.eslintcache/"
```

#### `--cache-strategy`

Strategy for the cache to use for detecting changed files.

* **Argument Type**: String. One of the following values:
  1. `metadata`
  1. `content`
* **Multiple Arguments**: No
* **Default Value**: `metadata`

> The `content` strategy can be useful in cases where the modification time of your files change even if their contents have not. For example, this can happen during git operations like `git clone` because git does not track file modification time.

Reviewer 4: ```suggestion
The `content` strategy can be useful in cases where the modification time of your files changes even if their contents have not. For example, this can happen during git operations like `git clone` because git does not track file modification time.
```

* **Argument Type**: No argument.

##### `--no-error-on-unmatched-pattern` example

```shell
npx eslint --no-error-on-unmatched-pattern --ext .ts "lib/*"
```

#### `--exit-on-fatal-error`

This option causes ESLint to exit with exit code 2 if one or more fatal parsing errors occur. Without this option, ESLint reports fatal parsing errors as rule violations.

* **Argument Type**: No argument.

##### `--exit-on-fatal-error` example

```shell
npx eslint --exit-on-fatal-error file.js
```

#### `--debug`

This option outputs debugging information to the console. Add this flag to an ESLint command line invocation in order to get extra debugging information while the command runs.

* **Argument Type**: No argument.

This information is useful when you're seeing a problem and having a hard time pinpointing it. The ESLint team may ask for this debugging information to help solve bugs.

##### `--debug` example

```shell
npx eslint --debug test.js
```

#### `--help`, `-h`

This option outputs the help menu, displaying all of the available options. All other options are ignored when this is present. When you use this flag, the CLI does not perform linting.

* **Argument Type**: No argument.

> ##### `--help` example

Reviewer: ```suggestion
##### `--help`, `-h` example
```

#### `--init`

> This option runs `npm init @eslint/config` to start config initialization wizard. It's designed to help new users quickly create `.eslintrc` file by answering a few questions. When you use this flag, the CLI does not perform linting.

Reviewer 4: ```suggestion
This option runs `npm init @eslint/config` to start the config initialization wizard. It's designed to help new users quickly create a `.eslintrc` file by answering a few questions. When you use this flag, the CLI does not perform linting.
```

Author: ```suggestion
This option runs `npm init @eslint/config` to start the config initialization wizard. It's designed to help new users quickly create an `.eslintrc` file by answering a few questions. When you use this flag, the CLI does not perform linting.
```

* **Argument Type**: No argument.

This information is useful when you're seeing a problem and having a hard time pinpointing it. The ESLint team may ask for this debugging information to help solve bugs.

##### `--debug` example

```shell
npx eslint --debug test.js
```

#### `--help`

This option outputs the help menu, displaying all of the available options. All other options are ignored when this is present. When you use this flag, the CLI does not perform linting.

* **Alias**: `-h`
* **Argument Type**: No argument.

##### `--help` example

```shell
npx eslint --help
```

#### `--version`

This option outputs the current ESLint version onto the console. All other options are ignored when this is present. When you use this flag, the CLI does not perform linting.

* **Alias**: `-v`
* **Argument Type**: No argument.

##### `--version` example

```shell
npx eslint --version
```

#### `--print-config`

This option outputs the configuration to be used for the file passed. When present, no linting is performed and only config-related options are valid. When you use this flag, the CLI does not perform linting.

> * **Argument Type**: No argument.

Reviewer 2: ```suggestion
* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No
```

`--print-config` should have an argument.

##### `--debug` example

```shell
npx eslint --debug test.js
```

#### `-h`, `--help`

This option outputs the help menu, displaying all of the available options. All other options are ignored when this is present. When you use this flag, the CLI does not perform linting.

* **Argument Type**: No argument.

##### `-h`, `--help` example

```shell
npx eslint --help
```

#### `-v`, `--version`

This option outputs the current ESLint version onto the console. All other options are ignored when this is present. When you use this flag, the CLI does not perform linting.

* **Argument Type**: No argument.

##### `-v`, `--version` example

```shell
npx eslint --version
```

#### `--print-config`

This option outputs the configuration to be used for the file passed. When present, no linting is performed and only config-related options are valid. When you use this flag, the CLI does not perform linting.

* **Argument Type**: String. Path to file.
* **Multiple Arguments**: No

##### `--print-config` example

```shell
npx eslint --print-config file.js
```

> ## Exit codes

Reviewer: ```suggestion
## Exit Codes
```
